modules/misunderstood/src/views/full/index.tsx

```diff
@@ -86,8 +88,12 @@ export default class MisunderstoodMainView extends React.Component<Props, State>
     })
   }
 
-  updateEventsCounts = async (language?: string, dateRange?: DateRange) => {
-    const eventCounts = await this.fetchEventCounts(language || this.state.language, dateRange || this.state.dateRange)
+  updateEventsCounts = async (language?: string, dateRange?: DateRange, reason?: FLAG_REASON) => {
+    const eventCounts = await this.fetchEventCounts(
+      language || this.state.language,
+      dateRange || this.state.dateRange,
+      reason != undefined ? reason : this.state.reason
```

Comment:
!==

Also, not sure why you check if the reason parameter is set, but use the reason from the state... Couldn't this line be replaced simply with `reason` ?

Reply:
reason argument could be null, if its null then the reason is not going to be filtered, but if its undefined then the state reason will be used, I will fix the triple equal, thanks